### DISCUSSION TITLE
[multitude of additions] Adding the Current storage of the VM in use and its limit.

### DISCUSSION THREAD

**Discussion by cloudrack-ca** [Performave/convoy#85] [Feature Requests & Suggestions]
- Adding the Current storage of the VM in use.
you have RAM specs CPU specs graphs and more but you're still missing how much of the drive is used out of x% permitted

- I Would also like to see the ability to add more than one drive to a VM and manage said drive
(solusvm | vmmanager | many others linode digitalocean and so on offer this would be nice)

- lxc usage at one point would be nice

- firewall management via the VM
(Pretty much most panels/hosts offer a means of some firewall'ing.)

- rDNS - bunny.net provides this CloudDNS and many others idea from (https://docs.virtfusion.com/administration/reverse-dns/) 

- new server button in the frontend panel for - admin users only of course.

- s3 backups (idea is to take the backup that was backuped via the proxmox server and send that file off to the s3 bucket? Possible idea many overthinking it for simplicity.

- white labling paid plan options, (rebrand options in settings.)

- Locking servers from being deleted convoypanel end or proxmox / both

more ideas but later on seems like you got a lot going on and are already doing quick and amazing work. also noticed maybe just my server itself at home here but I put my ipv4's/6 into ipam on convoypanel I click the breadcrumb example image below
![image](https://github.com/ConvoyPanel/panel/assets/145787423/b5ce820f-ab36-4ee7-97a9-f9120bf5008d)
I click on 10.0.0.2 in breadcrumb links I get the error below

![image](https://github.com/ConvoyPanel/panel/assets/145787423/c23cce90-e4d4-4228-8881-25af7b464468)

if its hard to read it states

```
Unhandled Thrown Error!
Request failed with status code 404
Error: Request failed with status code 404
    at cp (https://convoy.cloudrack.ca/build/assets/main-512c705b.js:103:193027)
    at fp (https://convoy.cloudrack.ca/build/assets/main-512c705b.js:103:193210)
    at XMLHttpRequest.$ (https://convoy.cloudrack.ca/build/assets/main-512c705b.js:104:1979)
```
💿 Hey developer 👋

You can provide a way better UX than this when your app throws errors by providing your own errorElement props on `<Route>`

**Comment by cloudrack-ca** [comment]
- Locking the ability to change a lower disk space then what was originally created for the VM i.e VM given 100GB admin can not set a VM storage lower then the 100GB.
- vCores + CPU limit + Numa Allowment + Socket Count. (changeable per a VM - as these are doable inside of proxmox itself should be doable) oh extra CPU flags too ( if possible )
- Ability to detect DHCPv6 Given IPv6 Addressess. (Once a VM Is created in this matter more less any VM inside of proxmox itself its shown the IP's of the VM in statistics for the VM (under summary)

- ability to delete broken or incomplete backups 
![image](https://github.com/ConvoyPanel/panel/assets/145787423/93a25096-61a8-4b05-8e10-a881d340b6f1)

- Ability to use Monitoring tools? or at least for time being view task history?

- when a user changes their DNS/IP/Hostname have a popup or notification stating to the user to `Restart their VM` when the changes are done for cloud-init to roll the changes on a reboot)

**Reply by cloudrack-ca** [reply]
@ericwang401 just some thoughts and ideas / suggestions for ya + improvements and bugs I've noticed.

**Reply by ericwang401** [reply]
Hi,

I'll convert your bullet points into a numbered list.

1. This is impossible since Proxmox doesn't allow anyone to do this. If this were possible, there would also be a great risk of data loss because the portion of the disk that is being deleted may contain data.
2. I need input from others on how this should be implemented.
3. I don't know how this would work?
4. Sure, please open an issue for this.
5. Something like this would be planned for v5. I'm planning on a UI overhaul because right now it's really hard to add new code to the UI.
6. Also sure, please open an issue for this too.

**Reply by cloudrack-ca** [reply]
also lastly i noticed UI issuing with 100000x ips when creating a VM and there is more then x amount im not sure the total to be honest, it doesn't allow me to select a ipv4 if I have 100000x ipv6 being pulled first.

**Reply by ericwang401** [reply]
I would not recommend creating 100,000 IPs in IPAM. This is a limitation of Convoy.

**Reply by cloudrack-ca** [reply]
no no i get it im just poking around thats all sorry.
